libethereum/BlockChain.cpp

```diff
@@ -1187,42 +1205,56 @@ void BlockChain::garbageCollect(bool _force)
 
 	m_lastCollection = chrono::system_clock::now();
 
-	Guard l(x_cacheUsage);
-	WriteGuard l1(x_blocks);
-	WriteGuard l2(x_details);
-	WriteGuard l3(x_blockHashes);
-	WriteGuard l4(x_receipts);
-	WriteGuard l5(x_logBlooms);
-	WriteGuard l6(x_transactionAddresses);
-	WriteGuard l7(x_blocksBlooms);
 	for (CacheID const& id: m_cacheUsage.back())
```

Comment:
But you need `x_cacheUsage` be read-locked for this

Reply:
OK.  I put back the guard (some lines above, might not be visible).